src/Preprocessor.zig

```diff
@@ -1281,9 +1282,8 @@ fn expandMacroExhaustive(
                     const count = macro_scan_idx - idx + 1;
                     for (buf.items[idx .. idx + count]) |tok| Token.free(tok.expansion_locs, pp.comp.gpa);
                     try buf.replaceRange(idx, count, res.items);
-                    // TODO: moving_end_idx += res.items.len - (macro_scan_idx-idx+1)
-                    // doesn't work when the RHS is negative (unsigned!)
-                    moving_end_idx = moving_end_idx + res.items.len - count;
+
+                    moving_end_idx = (moving_end_idx + res.items.len) -| count;
```

Comment:
This looks good but I want to document it so that I can understand it in the future as well.

My understanding is that `count` should be called `tokens_removed` and `res.items.len` should be called `tokens_added`.  So the easiest to understand version would be
```suggestion
                    moving_end_idx += tokens_added;
                    // Overflow here means that we encountered an unterminated argument list
                    // while expanding the body of this macro.
                    moving_end_idx -|= tokens_removed;
```